### PR TITLE
V22F-890 add sim car

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Adds a sim car so that SDALib can call the simulation functions, otherwise this cannot work for other bbs

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/BlackBoxData.cpp [hunk 1]
```diff
@@ -17,9 +17,10 @@
 /// @param p_tickCount The tick that this driving simulation is from
 /// @param p_nextSegments      Segment pointer to store the copied segments in
 /// @param p_nextSegmentsCount Amount of segments to copy
-BlackBoxData::BlackBoxData(tCarElt* p_car, tSituation* p_situation, unsigned long p_tickCount, tTrackSeg* p_nextSegments, int p_nextSegmentsCount)
-        : Car({}), Situation({}), TickCount(p_tickCount)
+BlackBoxData::BlackBoxData(tCar* p_carTable, tCarElt* p_car, tSituation* p_situation, unsigned long p_tickCount, tTrackSeg* p_nextSegments, int p_nextSegmentsCount)
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/BlackBoxData.cpp:20]: Update documentation

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl [hunk 2]
```diff
@@ -141,6 +142,14 @@ void TEMP_DECISIONMAKER::SetDataCollectionSettings(tDataToStore p_dataSetting)
     Config.SetDataCollectionSettings(p_dataSetting);
 }
 
+/// @brief          Sets the car table of the simulator
+/// @param p_compressionLevel The car table of the simulator
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl:146]: Param name incorrect

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.inl [hunk 3]
```diff
@@ -128,6 +129,14 @@ void Mediator<DecisionMaker>::SetCompressionRate(int p_compressionRate)
     m_decisionMaker.Config.SetCompressionRate(p_compressionRate);
 }
 
+/// @brief          Sets the car table of the simulator
+/// @param p_compressionLevel The car table of the simulator
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.inl:133]: Param name incorrect

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SocketBlackBox.inl [hunk 4]
```diff
@@ -246,15 +246,15 @@ void SocketBlackBox<BlackBoxData, PointerManager>::DeserializeBlackBoxResults(co
 /// @param p_decisions Decision tuple to store decisions in
 /// @return returns true if a decision is made, else false (will always return true if connection is not async)
 template <class BlackBoxData, class PointerManager>
-bool SocketBlackBox<BlackBoxData, PointerManager>::GetDecisions(tCarElt* p_car, tSituation* p_situation, unsigned long p_tickCount, DecisionTuple& p_decisions)
+bool SocketBlackBox<BlackBoxData, PointerManager>::GetDecisions(tCar* p_carTable, tCarElt* p_car, tSituation* p_situation, unsigned long p_tickCount, DecisionTuple& p_decisions)
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SocketBlackBox.inl:249]: Update documentation

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Add to Changes.txt and code changes document

**Review by suppergerrie2 — APPROVED** [review]

**Review by larsdekwant — CHANGES_REQUESTED** [review]
Cant connect with the SDALib-Python.exe black box
```
Connecting sync
D:\Universiteit\UU\Jaar-3\Software-Project\code\DAISI\speed-dreams\source-2.2.3\data
  File "<string>", line 1
    sys.path.append("D:\Universiteit\UU\Jaar-3\Software-Project\code\DAISI\speed-dreams\source-2.2.3\data")
                                                                                                          ^
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape
D:\Universiteit\UU\Jaar-3\Software-Project\code\DAISI\speed-dreams\source-2.2.3\data
  File "<string>", line 1
    sys.path.append("D:\Universiteit\UU\Jaar-3\Software-Project\code\DAISI\speed-dreams\source-2.2.3\data")
                                                                                                          ^
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape
Trying to connect to Speed Dreams
[WSA] Failed to receive message. Error code: 10054
[WSA] Connection error. Error code: 10054
[WSA] Failed to receive message. Error code: 10054
00:02:05.933 Default  Warning SLOW BLACKBOX: Blackbox took 250+ milliseconds (on average) to respond over 5 tests, this is relatively slow!
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/rppUtils/RppUtils.hpp [hunk 5]
```diff
@@ -206,8 +206,11 @@ inline void StartExecutable(const std::string& p_executablePath, const char* p_a
     // WARNING: This method of starting a process is Windows-exclusive.
     // Add a different method to run a process here if a Linux build is planned.
 
+    const char* workingDir = filesystem::path(p_executablePath).parent_path().string().c_str();
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/rppUtils/RppUtils.hpp:209]: This access freed memory, the string is deleted after this line is over.

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/rppUtils/RppUtils.hpp (hunk 5 above)
Comment (KungFuDonkey) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/rppUtils/RppUtils.hpp:209]: can be removed as it is not needed anymore

**Review by suppergerrie2 — APPROVED** [review]

**Review by larsdekwant — APPROVED** [review]

**Pull request merged**